<!DOCTYPE html>
<html>
<script>
    // try…catch…finally
    try {
        alert('try');
        if (confirm('Make an error?')) BAD_CODE();
    } catch (err) {
        alert('catch');
    } finally {
        alert('finally');
    }
    // ----------------
    let num = +prompt("Enter a positive integer number?", 35)

    let diff, result;

    function fib(n) {
        if (n < 0 || Math.trunc(n) != n) {
            throw new Error("Must not be negative, and also an integer.");
        }
        return n <= 1 ? n : fib(n - 1) + fib(n - 2);
    }

    let start = Date.now();

    try {
        result = fib(num);
    } catch (err) {
        result = 0;
    } finally {
        diff = Date.now() - start;
    }

    alert(result || "error occurred");

    alert(`execution took ${diff}ms`);

    // Global catch
    window.onerror = function (message, url, line, col, error) {
        alert(`${message}\n At ${line}:${col} of ${url}`)
    };
    function readData(){
        badFunc();
    }
    readData();
</script>

</html>